internal/utils/filter/matching.go

```diff
@@ -63,3 +44,39 @@ func stripNegation(value string) (string, bool) {
 	}
 	return value, false
 }
+
+type matcher interface {
+	// Returns whether the string matched and whether it was a negated match
+	Matches(s string) (bool, bool)
+}
+
+type staticMatcherSet map[string]bool
+
+var _ matcher = &staticMatcherSet{}
+
+func (m staticMatcherSet) Matches(s string) (bool, bool) {
+	negated, ok := m[s]
+	return ok, negated
+}
+
+type regexMatcher struct {
+	re      *regexp.Regexp
+	negated bool
+}
+
+var _ matcher = &regexMatcher{}
```

Comment:
why is this and the other ones assigned to _ needed?

Reply:
That is an interface assertion so that it fails to compile if it doesn't really implement the interface.